test/plugin/base_test.rb

```diff
@@ -605,6 +605,66 @@ def test_sanitize_tags_from_container_name_with_require_valid_tags_false
     end
   end
 
+  def test_configure_split_logs_by_tag
+    setup_gce_metadata_stubs
+    {
+      APPLICATION_DEFAULT_CONFIG => true,
+      DISABLE_SPLIT_LOGS_BY_TAG_CONFIG => false
+    }.each do |(config, split_logs_by_tag)|
+      d = create_driver(config)
+      assert_equal split_logs_by_tag,
+                   d.instance.instance_variable_get(:@split_logs_by_tag)
+    end
+  end
+
+  def test_split_logs_by_tag
+    setup_gce_metadata_stubs
+    log_entries_count = 5
+    [
+      [APPLICATION_DEFAULT_CONFIG, log_entries_count, log_entries_count],
+      [DISABLE_SPLIT_LOGS_BY_TAG_CONFIG, 1, log_entries_count]
+    ].each do |(config, successful_requests_count, ingested_entries_count)|
+      setup_prometheus
+      setup_logging_stubs do
+        d = create_driver(config + PROMETHEUS_ENABLE_CONFIG, 'test', true)
+        log_entries_count.times do |i|
+          d.emit("tag#{i}", 'message' => log_entry(0))
+        end
+        d.run
+      end
+      assert_prometheus_metric_value(:stackdriver_successful_requests_count,
```

Comment:
Should we not just check `@logs_sent` (the actual requests) rather than the Prometheus counter values?

Reply:
Added some checks. In fact, I just combined the two tests `test_split_logs_by_tag` and `test_log_name_when_split_logs_by_tag_disabled`.